packages/gatsby/src/schema/node-model.js

```diff
@@ -785,6 +787,11 @@ const determineResolvableFields = (
     if (!fieldsToResolve[fieldName] && needsResolve) {
       fieldsToResolve[fieldName] = true
     }
+    if (!fieldsToResolve[fieldName] && isNestedType) {
+      // If parent field needs to be resolved - all nested fields should be added as well
+      // See https://github.com/gatsbyjs/gatsby/issues/26056
+      fieldsToResolve[fieldName] = true
+    }
```

Comment:
Small nit
```suggestion
    if (!fieldsToResolve[fieldName] && (needsResolve || isNestedType)) {
      // If parent field needs to be resolved - all nested fields should be added as well
      // See https://github.com/gatsbyjs/gatsby/issues/26056
      fieldsToResolve[fieldName] = true
    }
```

Reply:
I thought about this but then the comment looks like it also relates to the `needsResolve` part which is not the case.

Reply:
true, let's keep it as is